tests/upgrades_tests.cairo

```diff
@@ -0,0 +1,114 @@
+use traits::Into;
+use traits::TryInto;
+
+use starknet::{ContractAddress, storage_access::storage_address_from_base};
+
+use konoha::contract::IGovernanceDispatcher;
+use konoha::contract::IGovernanceDispatcherTrait;
+use konoha::proposals::IProposalsDispatcher;
+use konoha::proposals::IProposalsDispatcherTrait;
+use konoha::upgrades::IUpgradesDispatcher;
+use konoha::upgrades::IUpgradesDispatcherTrait;
+use konoha::constants;
+
+use snforge_std as snf;
+use snforge_std::{CheatTarget, ContractClassTrait};
+
+use super::setup::{admin_addr, deploy_governance, deploy_and_distribute_gov_tokens,};
+
+
+#[test]
+fn test_apply_a_passed_proposal() {
+    let token_contract = deploy_and_distribute_gov_tokens(admin_addr.try_into().unwrap());
+    let gov_contract = deploy_governance(token_contract.contract_address);
+    let gov_contract_addr = gov_contract.contract_address;
+    let dispatcher = IProposalsDispatcher { contract_address: gov_contract_addr };
+    let class_hash: felt252 = snf::get_class_hash(token_contract.contract_address)
+        .try_into()
+        .unwrap();
+
+    assert!(
+        snf::get_class_hash(gov_contract_addr).try_into().unwrap() != class_hash,
+        "Wrong initial class hash"
+    );
+
+    snf::start_prank(CheatTarget::One(gov_contract_addr), admin_addr.try_into().unwrap());
```

Comment:
I prefer prank() instead of start_prank because if you don't call stop_prank later and call start_prank again on the same contract, it doesn't override the call, which can lead to some hard to debug bugs.

Reply:
Updated